web/packages/teleport/src/Users/UserAddEdit/useDialog.tsx

```diff
@@ -32,6 +34,9 @@ export default function useUserDialog(props: Props) {
       label: r,
     }))
   );
+  const [configuredTraits, setConfiguredTraits] = useState<TraitsOption[]>(
+    traitsToTraitsOption(props.user.allTraits)
+  );
```

Comment:
otherwise, `traitsToTraits` func will needlessly run every re-render
```suggestion
  const [configuredTraits, setConfiguredTraits] = useState<TraitsOption[]>(() =>
    traitsToTraitsOption(props.user.allTraits)
  );
```